plugins/inventory/azure_rm.py

```diff
@@ -463,30 +472,61 @@ def _process_queue_batch(self):
                     # FUTURE: error-tolerant operation mode (eg, permissions)
                     # FUTURE: store/handle errors from individual handlers
                     result.handler(r['content'], **result.handler_args)
+                    batch_processed.append(returned_name)
+                elif status_code in _RETRY_CODES:
+                    # 429: Too many requests Error, Backoff and Retry
+                    batch_retry = True
+            if batch_retry:
+                time.sleep(backoff_factor * (2 ** (retry_count)))
```

Comment:
Regarding the backoff policy, retry time should based on the response. Other wise the retry request will be rejected too.
https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/windows/troubleshooting-throttling-errors#throttling-error-details

Reply:
Thank you @xuzhang3,
@p3ck is on vacation (he will be back next week) so I'm checking how to add the RetryPolicy class to be used here, in order to have this response retry time to be considered

Reply:
The issue here is that in the current implementation the wait sleep is done once for the all batch together
Do you have any suggestion/thoughts how to change it to work for each vm separately?

Reply:
@nirarg  I see, misunderstand the retry policy here.  The backoff policy and retry is focus on the local batch requests here so we cannot determine the retry time for all the request.